### ISSUE TITLE
Consider filtering options for the ListTasks endpoint

### ISSUE THREAD

**Issue by buchanae** [ga4gh/task-execution-schemas#57]
Currently the ListTasksRequest contains a couple fields for filtering
```
message ListTasksRequest {

  // OPTIONAL
  //
  // Filter the task list to include tasks in this project.
  string project = 1;

  // OPTIONAL
  //
  // Filter the list to include tasks where the name matches this prefix.
  // If unspecified, no task name filtering is done.
string name_prefix = 2;

...
```

There have been suggestions for changing this:
- Remove `project` and `name_prefix`
- Allow filtering on Task.state and Task.tags

**Comment by buchanae** [comment]
#92 is where most of the work is happening

**Comment by adamstruck** [comment]
The current proposal is reflected in #104. This approach would require a switch to OpenApi 3.0

**Comment by vsmalladi** [comment]
@kellrott has this been addressed by #170
